app/src/main/java/de/danoeh/antennapod/fragment/CoverFragment.java

```diff
@@ -184,20 +184,35 @@ public float convertDpToPixel(float dp) {
 
     private void configureForOrientation(Configuration newConfig) {
         LinearLayout mainContainer = getView().findViewById(R.id.cover_fragment);
-        ViewGroup.LayoutParams params = imgvCover.getLayoutParams();
+        LinearLayout textContainer = getView().findViewById(R.id.cover_fragment_text_container);
+
+        LinearLayout.LayoutParams params = (LinearLayout.LayoutParams) imgvCover.getLayoutParams();
+        LinearLayout.LayoutParams textParams = (LinearLayout.LayoutParams) textContainer.getLayoutParams();
+
 
         if (newConfig.orientation == Configuration.ORIENTATION_PORTRAIT) {
+            double ratio = (float) newConfig.screenHeightDp / (float) newConfig.screenWidthDp;
+            double percentageWidth = 0.8;
+            if (ratio <= 1.7) {
+                percentageWidth = (ratio / 1.7) * percentageWidth * 0.8;
+            }
             mainContainer.setOrientation(LinearLayout.VERTICAL);
             if (newConfig.screenWidthDp > 0) {
-                params.width = (int) (convertDpToPixel(newConfig.screenWidthDp) * .80);
+                params.width = (int) (convertDpToPixel(newConfig.screenWidthDp) * percentageWidth);
                 params.height = params.width;
+                textParams.weight = 0;
                 imgvCover.setLayoutParams(params);
             }
         } else {
+            double ratio = (float) newConfig.screenHeightDp / (float) newConfig.screenWidthDp;
+            double percentageHeight = 0.4;
+            percentageHeight = (ratio / 0.5) * percentageHeight;
```

Comment:
Where does the 0.5 come from? Wouldn't this be the same as multiplying with 2?

Reply:
I have to spend a bit of time thinking of how to do the calculation.  When do you need this work to be done for 2.0 release?

Reply:
As early as possible, I would say. I hope that the last big thing holding back 2.0.0 (new icon poll) is done in about 1-2 weeks.

Reply:
see new code pushed